### ISSUE TITLE
Telling libclang where to look for system includes

### ISSUE THREAD

**Issue by dylanede** [rust-lang-nursery/rust-bindgen#975]
I've had success using bindgen on Windows with MSVC, but in order to do so I have to specify the `INCLUDE` environment variable as so (minus new-lines):

```text
C:\Program Files (x86)\Microsoft Visual Studio\Shared\14.0\VC\include;
C:\Program Files (x86)\Windows Kits\10\Include\10.0.10240.0\ucrt;
C:\Program Files (x86)\Windows Kits\10\Include\10.0.10240.0\um;
C:\Program Files (x86)\Windows Kits\10\Include\10.0.10240.0\shared
```

Without this definition clang fails to find the system headers. I also have to specify `-fms-compatibility-version=19` (not 14) via `Builder::clang_arg` to get this to work.

Obviously this configuration is specific to this particular version of MSVC and my installation setup, so I'm reluctant to hard-code this into a crate that uses bindgen. Ideally bindgen would be able to, via use of the gcc crate, figure out whether additional configuration of libclang is necessary to match the system headers and ABI of the toolchain decided upon by the gcc crate (this allows bindgen output to match the ABI of, for example, external C++ code built with the cmake crate).

In terms of achieving this, it appears straightforward when the C++ compiler picked is gcc or clang, as they can be queried easily (via a flag) for the system include paths. It looks to be much harder to get this information out of MSVC. There are two "IDE" variables that contribute to the system include directories list by default, `VC_IncludePath` and `WindowsSDK_IncludePath`. Their definitions are in separate XML files whose locations vary based on MSVC version and installation. The most robust way of finding their values may be to have a MSVC project (perhaps generated using cmake) that builds an executable that prints them out.

Thoughts? Please note that implementing at least part of this would be of benefit when using gcc or clang, not just MSVC, as it would ensure bindgen matches the standard library implementation.

**Comment by fitzgen** [comment]
I agree that this can be a pain, but I don't feel like I personally understand msvc/windows development enough to fully understand the different options available. (Or even an understanding of all the platform differences and little complexities.)

We do have some auto path stuff here: https://github.com/rust-lang-nursery/rust-bindgen/blob/bb315bfa7c23333a2c02374d7b7dbfbd3a97f0b6/src/lib.rs#L1411-L1436

Undoubtedly it is incomplete and/or incorrect in some ways.

Perhaps @emilio has a better grasp of these things.

**Comment by fitzgen** [comment]
Apparently the `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Microsoft SDKs\Windows` registry key has a subkey for each installed SDK, with an `InstallationFolder` value. And then the include directory should be something like `./Include` from that. Except when it is dependent on the build number. In which case we can maybe crawl that corner of the filesystem.

**Comment by AndrewGaspar** [comment]
It would be nice if bindgen had the same behavior for looking up VC++ installs and Windows SDKs as the `cc` crate: https://github.com/alexcrichton/cc-rs/blob/master/src/windows_registry.rs
